src/main/java/frc/robot/RobotContainer.java

```diff
@@ -158,55 +124,41 @@ private void defineButtons() {
    * edu.wpi.first.wpilibj2.command.button.JoystickButton}.
    */
   private void configureButtonBindings() {
-    if (Constants.HARDWARE_CONFIG_HAS_DRIVETRAIN) {
+    if (drivetrainSubsystem != null) {
       // Back button zeros the gyroscope
-      resetGyro.whenPressed(drivetrainSubsystem::zeroGyroscope);
-    }
+      resetGyro.whenPressed(() -> drivetrainSubsystem.zeroGyroscope());
```

Comment:
Is this just a style change or is there a technical reason for it?

Reply:
It was showing as an error in my VS Code the other way, probably should have tried to build. Both ways should be fine.

Reply:
I need to see how this looks when deployed. On paper (screen?) this makes sense, but I'm not sure it will work when implemented. The purpose of the global bools was to make it easier to exclude physical subsystems that might be missing/not present when prototyping code for other subsystems without needing to comment out a bunch of code in RobotContainer. 

The specific error(s) we are looking to avoid are CAN ID errors which may prevent enabling the robot. If a subsystem object is declared at the top of the class it will be null, but if we continue on to construct the object when that CAN ID is not present on the robot, then it will error after deploying. At least that is how I think it works.

Approving PR  as this is easily fixed later

Reply:
I didn't remove the global bools, I am still using those for initialization.

The error here was a syntax error (something about the function being void not supported)